### PR TITLE
Feature/ticket search

### PR DESCRIPTION
**Pull request by priyankatechneai**
ticket-search functionality

### CODE REVIEW

File: src/screens/TicketManagement/MyTicketComponent.jsx [hunk 1]
```diff
@@ -261,10 +265,9 @@ export default function MyTicketComponent() {
                       <i className="icofont-ui-edit"></i> Edit
```
Comment (somyaranjan-techneai) [inline src/screens/TicketManagement/MyTicketComponent.jsx:265]: make self-closing tags

File: src/screens/TicketManagement/MyTicketComponent.jsx [hunk 2]
```diff
@@ -307,7 +306,7 @@ export default function MyTicketComponent() {
                       <i className="icofont-tasks"></i> Task
```
Comment (somyaranjan-techneai) [inline src/screens/TicketManagement/MyTicketComponent.jsx:306]: make self-closing tags

File: src/screens/TicketManagement/MyTicketComponent.jsx [hunk 3]
```diff
@@ -274,12 +277,10 @@ export default function MyTicketComponent() {
                 </Link>{" "}
               </li>
 
-              {data.created_by != localStorage.getItem("id") &&
-                data.basket_configured === 0 &&
-                localStorage.getItem("account_for") === "SELF" &&
-                data.status_name != "Solved" &&
-                data.passed_status !== "REJECT" &&
-                data.passed_status !== "UNPASS" && (
+              {((data.created_by != UserId && data.basket_configured === 0) ||
```
Comment (AmitSolnke) [inline src/screens/TicketManagement/MyTicketComponent.jsx:280]: No optional chain

Comment (AmitSolnke) [inline src/screens/TicketManagement/MyTicketComponent.jsx:280]: Again no use of strict equal operator

File: src/screens/TicketManagement/MyTicketComponent.jsx [hunk 4]
```diff
@@ -291,13 +292,11 @@ export default function MyTicketComponent() {
...
                 )}
 
-              {(data.created_by != localStorage.getItem("id") &&
-                data.basket_configured > 0 &&
-                data.status_name != "Solved" &&
-                localStorage.getItem("account_for" === "SELF")) ||
-                (data?.projectowner?.filter(
-                  (d) => d.user_id == localStorage.getItem("id")
-                ) && (
+              {((data.created_by != localStorage.getItem("id") &&
```
Comment (AmitSolnke) [inline src/screens/TicketManagement/MyTicketComponent.jsx:295]: Same here, no strict operation no optional chain

File: src/screens/TicketManagement/MyTicketComponent.jsx [hunk 5]
```diff
@@ -318,6 +317,18 @@ export default function MyTicketComponent() {
                   <i className="icofont-history"></i> History
                 </Link>
               </li>
+
+              {data.created_by === Number(UserId) && (
```
Comment (AmitSolnke) [inline src/screens/TicketManagement/MyTicketComponent.jsx:321]: Same here

File: src/screens/TicketManagement/MyTicketComponent.jsx (hunk 1 above)
Comment (priyankatechneai) [reply src/screens/TicketManagement/MyTicketComponent.jsx:265]: resolved

File: src/screens/TicketManagement/MyTicketComponent.jsx (hunk 2 above)
Comment (priyankatechneai) [reply src/screens/TicketManagement/MyTicketComponent.jsx:306]: resolved

File: src/screens/TicketManagement/MyTicketComponent.jsx [hunk 6]
```diff
@@ -318,6 +317,18 @@ export default function MyTicketComponent() {
                   <i className="icofont-history"></i> History
```
[the remark this answers is not in the document]
Comment (priyankatechneai) [reply src/screens/TicketManagement/MyTicketComponent.jsx:317]: resolved

File: src/screens/TicketManagement/MyTicketComponent.jsx (hunk 3 above)
Comment (priyankatechneai) [reply src/screens/TicketManagement/MyTicketComponent.jsx:280]: resolved

File: src/screens/TicketManagement/MyTicketComponent.jsx (hunk 5 above)
Comment (priyankatechneai) [reply src/screens/TicketManagement/MyTicketComponent.jsx:321]: resolved

File: src/screens/TicketManagement/MyTicketComponent.jsx (hunk 3 above)
Comment (priyankatechneai) [reply src/screens/TicketManagement/MyTicketComponent.jsx]: resolve

**Comment by priyankatechneai** [comment]
resolved

**Pull request merged**